### PR TITLE
Split incoming initial and trailing metadata in Ruby calls

### LINKED ISSUE #7249: Split received initial and trailing metadata in Ruby API
These two things are semantically distinct, and should be exposed separately.

### PR DESCRIPTION
**Pull request by murgatroid99**
This fixes #7249. Also, removes an unnecessary line from ActiveCall status initialization, making metadata representation more consistent between error and other contexts. This is an API breakage because trailing metadata no longer appears in the `metadata` accessor.

### CODE REVIEW

File: src/ruby/lib/grpc/generic/active_call.rb [hunk 1]
```diff
@@ -456,17 +454,19 @@ def initialize(wrapped)
     # SingleReqView limits access to an ActiveCall's methods for use in server
     # handlers that receive just one request.
     SingleReqView = view_class(:cancelled, :deadline, :metadata,
```
Comment (apolcyn) [inline src/ruby/lib/grpc/generic/active_call.rb]: Thought that SingleReqView and MultiReqView were only for call objects in server handlers, in which case trailing metadata shouldn't be relevant?

Comment (murgatroid99) [inline src/ruby/lib/grpc/generic/active_call.rb]: You're right. I've reverted those changes.

File: src/ruby/spec/generic/rpc_server_spec.rb [hunk 2]
```diff
@@ -95,7 +95,7 @@ class FailingService
   def initialize(_default_var = 'ignored')
     @details = 'app error'
     @code = 101
-    @md = { failed_method: 'an_rpc' }
+    @md = { 'failed_method' => 'an_rpc' }
```
Comment (apolcyn) [inline src/ruby/spec/generic/rpc_server_spec.rb]: Do hash keys sent from one end show up only as a hash key of that type on the other end?

Comment (murgatroid99) [inline src/ruby/spec/generic/rpc_server_spec.rb]: No, they always arrive as strings.

**Comment by apolcyn** [comment]
Not sure why the builds are failing, looked to be related to git. Otherwise LGTM

**Comment by murgatroid99** [comment]
test this please

**Comment by murgatroid99** [comment]
OK, Ruby tests are green now

**Pull request merged**